### PR TITLE
#1-4Task. dmitry.mukhliada

### PR DESCRIPTION
**Pull request by N0menIllisLegio**
https://travis-ci.org/github/N0menIllisLegio/js-assignments/builds/709765352

### CODE REVIEW

File: task/02-numbers-tasks.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ function getAverage(value1, value2) {
  *   (-5,0) (10,-10) => 18.027756377319946
  */
 function getDistanceBetweenPoints(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.sqrt(Math.pow((x2 - x1), 2) + Math.pow((y2 - y1), 2));
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js]: use math.hypot

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ function getLinearEquationRoot(a, b) {
  *   (0,1) (1,2)     => 0
  */
 function getAngleBetweenVectors(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.acos((x1 * x2 + y1 * y2) / (Math.sqrt(x1 * x1 + y1 * y1) * Math.sqrt(x2 * x2 + y2 * y2)));
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js]: use math.hypot

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -160,7 +160,7 @@ function parseNumberFromString(value) {
  *   1,2,3   => 3.741657386773941
  */
 function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+    return Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js]: use math.hypot

File: task/02-numbers-tasks.js [hunk 4]
```diff
@@ -202,7 +209,25 @@ function roundToPowerOfTen(num, pow) {
  *   17 => true
  */
 function isPrime(n) {
-    throw new Error('Not implemented');
+    if (n <= 1) {
+        return false;
+    } else if (n <= 3) {
+        return true;
+    } else if ((n % 2 === 0) || (n % 3 === 0)) {
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js]: you can skip **else**  if there a return statement in if clause above

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -519,7 +597,11 @@ function group(array, keySelector, valueSelector) {
  *   ['one','two','three'], x=>x.split('')  =>   ['o','n','e','t','w','o','t','h','r','e','e']
  */
 function selectMany(arr, childrenSelector) {
-    throw new Error('Not implemented');
+  let res = Array();
+    
+  arr.map(x => res = res.concat(childrenSelector(x)));
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js]: result of map function is not used. If you need to just iterate over the array do not use map. Try to use reduce

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -536,7 +614,9 @@ function selectMany(arr, childrenSelector) {
  *   [[[ 1, 2, 3]]], [ 0, 0, 1 ]      => 2        (arr[0][0][1])
  */
 function getElementByIndexes(arr, indexes) {
-    throw new Error('Not implemented');
+  indexes.map(x => arr = arr[x]);
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js]: do not modifiy arguments if not required.

Result of map funcitons is not used

File: task/04-arrays-tasks.js [hunk 7]
```diff
@@ -503,7 +573,15 @@ function distinct(arr) {
  *   }
  */
 function group(array, keySelector, valueSelector) {
-   throw new Error('Not implemented');
+  let map = new Map();
+
+  array.map(function(x) {
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js]: result of the map function is not used

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IlyaMokin** [comment]
Thanks, looks fine now, closing the issue

**Pull request closed without merging**